pilot/pkg/config/kube/gateway/model.go

```diff
@@ -35,6 +37,16 @@ const (
 	serviceTypeOverride          = "networking.istio.io/service-type"
 )
 
+var supportedFeatures = suite.AllFeatures
```

Comment:
For tests...

> I am interested in opinions about how to incorporate the tests for it. Adding a static list for a the golden.yaml files seems sub-optimal as we will need to change this every time we support a new feature.

Maybe we can override to 1-2 features in the test so we don't overload it. We don't really need to test that we write _all_ of them, just that we write any (presumably, if we write some we could write them all).

---

I am a bit worried about our code hardcoding AllFeatures. Should we have this somewhere that this code and our conformance test (tests/integration/pilot/gateway_conformance_test.go) read from? That way they cannot get out of sync

Reply:
Ultimately, and perhaps relatively soon after this is merged(or together with this change if you think its better), we want to take the list of supportedFeatures from GatewayClass, and based on this list run the conformance tests.

Regardless, I agree that this should live in somewhere appropriate in the codebase. I thought model.go is the file to go but perhaps I could add a new file. 
wdyt?